gulpfile.js

```diff
@@ -39,7 +39,7 @@ gulp.task('build-debug-infos', async () => {
 });
 
 gulp.task('build-source', async () => {
-    const cli = require.resolve('@cocos/build-engine/dist/cli');
+    const cli = require.resolve('@cocos/build-engine/lib/cli');
```

Comment:
Why change the url? Will it break compatibility?

Reply:
Turbo's build pipeline can implement build caching to improve build efficiency, provided that we need to configure the input and output of each module's build task

https://github.com/cocos/cocos-engine/blob/55693df4c4116a76af0ba29b83d45d7f89677e31/turbo.json#L4-L9

Here we unify the input of all modules into the src directory and the output into the lib directory

Reply:
it won't break compatibility, I also update the path deps for native-pack-tool here 
https://github.com/cocos/creator-runtime-extensions/pull/479